### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by filipsch**
@vincentvankrunkelsven can you have a quick look?
- Override of solution environment now possible: explicitly copies the solution environment and then 'decorates' or 'overwrites' it with user-defined stuff. The explicit copy is necessary so that the update of the environment does not persist via the `RootState`.
- Messaging has been cut up to use S3 functions.
- Also improvements to capitalization and trim.

### CODE REVIEW

File: R/check-that.R [hunk 1]
```diff
@@ -124,12 +124,11 @@ convert_formula <- function(form) {
   deparsed <- lapply(form, deparse)[2:n]
   target = deparsed[[1]]
   explan = deparsed[[2]]
-  explan = sort(trim(strsplit(explan, "\\+")[[1]]))
+  trm <- function(x) gsub("^\\s+|\\s+$", "", x)
+  explan = sort(trm(strsplit(explan, "\\+")[[1]]))
```
Comment (vincentvankrunkelsven) [inline R/check-that.R]: Why are you using both `=` and `<-` here?

File: R/state.R [hunk 2]
```diff
@@ -218,10 +218,44 @@ decorate_state <- function(state, stud, sol, el = NULL) {
...
+    }
+  }
   return(sub_state)
 }
+
+#' @rdname override
+#' @export
+override_solution_code <- function(state, code) {
+  override_solution(state, code = code)
+}
```
Comment (vincentvankrunkelsven) [inline R/state.R]: I don't really understand why you use the same function `override_solution` here while the logic for the two is completely split up.

Comment (filipsch) [inline R/state.R]: `override_solution_code()` and `override_solution_env()` both wrap around `override_solution()`, for convenience

File: R/check-that.R (hunk 1 above)
Comment (filipsch) [inline R/check-that.R]: style policed it.

**Pull request merged**